### PR TITLE
Scm fixes

### PR DESCRIPTION
**Pull request by srinivasupadhya**

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb [hunk 1]
```diff
@@ -166,6 +166,8 @@ def update(cruise_config)
 
           @pluggable_scm_service.validate(scm)
 
+          scm.clearEmptyConfigurations()
```
Comment (arvindsv) [inline server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb]: Is this about removing empty parts of the config? Meaning, fields which have been "emptied" by the user, in the config?

Is there a corresponding test for this? Is this why the optional-field was added, in the specs, in this commit?

Comment (srinivasupadhya) [inline server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb]: yes. `optional-field` would not be null if `clearEmptyConfigurations()` doesn't happen.

Comment (arvindsv) [inline server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb]: Ok.

**Comment by srinivasupadhya** [comment]
Done as part of #818

**Pull request merged**